src/components/common/form/GenericForm.tsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react'
+import useForm from 'common/form/useForm'
+import { FormContextProvider } from 'components/common/form/FormContext'
+
+export type GenericFormProps = {
+  onSubmit: any
+  initialValues?: any
+  validationSchema: any
+  children: any
+}
+
+const GenericForm = (props: GenericFormProps) => {
+  const { initialValues, onSubmit, validationSchema, children } = props
+  const { formik } = useForm({ initialValues, onSubmit, validationSchema })
+
+  return (
+    <form onSubmit={formik.handleSubmit}>
+      <FormContextProvider value={formik}>{children}</FormContextProvider>
+    </form>
+  )
+}
+
+export default GenericForm
```

Comment:
We can improve the typing by passing the `FormikConfig` to the component. This will gives several advantages combined with `...formProps`

- We don't have to type them manually
- We can pass more props to the `useForm` hook without changing this component
- We pass specific type to `useForm` and in this way the form knows what kind of values we drill down

By using `React.PropsWithChildren<{}>` we skip the need to specify the `children` type manually

I would make the context the wrapper of the actual `<form />` component to keep the semantics.

```suggestion
import { FormikConfig } from 'formik'
import React, { PropsWithChildren } from 'react'

import useForm from 'common/form/useForm'
import { FormContextProvider } from 'components/common/form/FormContext'

export type GenericFormProps<T> = PropsWithChildren<FormikConfig<T>>

export default function GenericForm<T>({ children, ...formProps }: GenericFormProps<T>) {
  const { formik } = useForm(formProps)

  return (
    <form onSubmit={formik.handleSubmit}>
      <FormContextProvider value={formik}>{children}</FormContextProvider>
    </form>
  )
}
```